packages/medusa/src/services/totals.js

```diff
@@ -72,39 +72,19 @@ class TotalsService extends BaseService {
    *    and applied discount
    */
   async getAllocationItemDiscounts(discount, cart) {
-    let discounts = []
+    const discounts = []
     for (const item of cart.items) {
-      discount.discount_rule.valid_for.forEach(async v => {
+      discount.discount_rule.valid_for.map(async v => {
```

Comment:
Check that `valid_for` exists otherwise we get map is not a function

Reply:
Even though default is set to `[]`?